nexus/plugins/repos.py

```diff
@@ -62,27 +63,25 @@ def copy_build_repo(self, host, conf_dict):
         source = self.build_repo_file
         destination = "/etc/yum.repos.d/" + source
 
-        logger.log.info("source file is %s" % source)
-        logger.log.info("destination file is %s" % destination)
-
-        ssh_c = SSHClient(hostname = host, username = \
-                                  self.username, password = self.password)
-        ssh_c.CopyFiles(source, destination)
+        dist = Platform(host, username, password)
+        distver = dist.replace('.','')
+        if distver in self.build_repo_file:
+            logger.log.info("source file is %s" % source)
+            logger.log.info("destination file is %s" % destination)
 
+            ssh_c = SSHClient(hostname = host, username = \
+                                      self.username, password = self.password)
+            ssh_c.CopyFiles(source, destination)
+        else:
+            logger.log.info("destination is %s" % dist)
+            logger.log.info("Not adding repo file to %s" % destination)
 
     def copy_async_updates_repo(self, host, conf_dict):
         """copy the async updates repo to all the existing nodes"""
 
         try:
             logger.log.info("Checking platform.dist of %s to get the right batched repo" % host)
-            ssh_c = SSHClient(hostname = host, username = \
-                                      self.username, password = self.password)
-            stdin, stdout, stderr = ssh_c.ExecuteCmd('python -c "import platform; \
-                                                     print platform.dist()"')
-            dist = stdout.read()
-            dist = str(dist).replace('(','').replace(')','').replace("'", "").\
-                   replace(',','')
-            dist = dist.split()
+            dist = Platform(host, username, password)
 
             logger.log.info("Platform distribution for host %s is %s" % (host, dist))
```

Comment:
Ah! forgot to mention. There are around 2-3 places like this where we log the platform.dist(). With only dist[1] returned these logs will look incomplete.